pkg/apis/numaplane/v1alpha1/status_types.go

```diff
@@ -117,9 +117,37 @@ type ChildStatus struct {
 	AssessUntil *metav1.Time `json:"assessUntil,omitempty"`
 }
 
+// ScaleValues stores the desired min and max, scaleTo, and actual scale values of a pipeline or monovertext vertex
+type ScaleValues struct {
+	// DesiredMin is the min scale value of the original child spec
+	DesiredMin int64 `json:"desiredMin"`
+	// DesiredMax is the max scale value of the original child spec
+	DesiredMax int64 `json:"desiredMax"`
+	// ScaleTo indicates how many pods to scale down to and it is calculated by getting the floor of the desiredMax value divided by 2
+	ScaleTo int64 `json:"scaleTo"`
+	// Actual indicates how many pods are actually running for the vertex
+	Actual int64 `json:"actual"`
+}
+
+// PromotedChildStatus describes the status of the promoted child
+type PromotedChildStatus struct {
+	// Name of the promoted child
+	Name string `json:"name"`
+	// ScaleValues is a map where the keys are the promoted child source vertices names
+	// and the values are the scale values of the source vertices
+	ScaleValues map[string]ScaleValues `json:"scaleValues,omitempty"`
+	// AllSourceVerticesScaledDown indicates if ALL the promoted child source vertices have been scaled down
+	AllSourceVerticesScaledDown bool `json:"allSourceVerticesScaledDown,omitempty"`
+	// ScaleValuesRestoredToDesired indicates if ALL the promoted child source vertices have been set back to the original min and max scale values (desiredMin and desiredMax).
+	// This field being set to `true` invalidates the value(s) in the scaleValues.Actual field.
+	ScaleValuesRestoredToDesired bool `json:"scaleValuesRestoredToDesired,omitempty"`
```

Comment:
If we are going to have all this scale stuff in our generic `PromotedChildStatus`, then let's at least have a struct to encapsulate it, and clearly comment that it only pertains to some Rollouts and not others

Reply:
I think we'll later need to explore the potential of not all Rollouts using the same `ProgressiveStatus` type. Maybe I can look at that.